keras/callbacks.py

```diff
@@ -830,41 +830,23 @@ def is_indexed_slices(grad):
 
             projector.visualize_embeddings(self.writer, config)
 
-    def on_epoch_end(self, epoch, logs=None):
-        logs = logs or {}
+    def on_epoch_begin(self, epoch, logs=None):
+        if self.model.test_function is None:
+            return
+        self.model.test_function._epoch = epoch
+        self.model.test_function._writer = self.writer
+        if self.histogram_freq and epoch % self.histogram_freq == 0:
+            self.model.test_function._merged_summaries = self.merged
+        else:
+            self.model.test_function._merged_summaries = None
 
+    def on_epoch_end(self, epoch, logs=None):
         if not self.validation_data and self.histogram_freq:
             raise ValueError('If printing histograms, validation_data must be '
```

Comment:
This message needs to be removed, as `validation_data` is no longer passed to the callback.

Reply:
`batch_size` parameter needs to be deprecated too.

Reply:
I thought since `callbacks.Tensorboard` didn't need `validation_data` anymore (and no other callback seems to use it), we could safely remove it; but I guess this would break users' custom callbacks. +1 for keeping it, then.

Reply:
It isn't used directly in the callback itself, but `test_loop` uses it. That is where the TensorBoard callback plugs in. Without validation data, it wouldn't be called.